File: Cpp/Kode/Guitar.cpp
#include "MusicInstrument.cpp"

class Guitar : public MusicInstrument{
    private :
        string brand;
        string series;
        int fretSize;
        string stringType;
    public :
        // constructor kosong
        Guitar(){
            brand = "";
            series = "";
            fretSize = 0;
            stringType = "";
        };
        // constructor berparameter (memanggil constructor MusicInstrument)
        Guitar(int id, double price, int stock,
               string playingMethod, string condition, string weight,
               string brand, string series, int fretSize, string stringType)
            : MusicInstrument(id, price, stock, playingMethod, condition, weight){
            this->brand = brand;
            this->series = series;
            this->fretSize = fretSize;
            this->stringType = stringType;
        }
        // getter and setter brand
        void setBrand(string brand){this->brand=brand;}
        string getBrand(){return brand;}

        // getter and setter series
        void setSeries(string series){this->series=series;}
        string getSeries(){return series;}

        // getter and setter fretSize
        void setFretSize(int fretSize){this->fretSize=fretSize;}
        int getFretSize(){return fretSize;}

        // getter and setter stringType
        void setStringType(string stringType){this->stringType=stringType;}
        string getStringType(){return stringType;}

        // destructor
        ~Guitar(){}
};

File: PHP/kode/Guitar.php
<?php
require_once 'MusicInstrument.php';

class Guitar extends MusicInstrument {
    private $brand='';
    private $series='';
    private $fretSize=0;
    private $stringType='';

    public function __construct($id = 0, $price = 0.0, $stock = 0,
                                 $playingMethod = "", $condition = "", $weight = "",
                                 $brand = "", $series = "", $fretSize = 0, $stringType = "",
                                 $photo = "") {
        parent::__construct($id, $price, $stock, $playingMethod, $condition, $weight, $photo);
        $this->brand = $brand;
        $this->series = $series;
        $this->fretSize = $fretSize;
        $this->stringType = $stringType;
    }

    // getter and setter brand
    public function setBrand($brand) { $this->brand = $brand; }
    public function getBrand() { return $this->brand; }

    // getter and setter series
    public function setSeries($series) { $this->series = $series; }
    public function getSeries() { return $this->series; }

    // getter and setter fretSize
    public function setFretSize($fretSize) { $this->fretSize = $fretSize; }
    public function getFretSize() { return $this->fretSize; }

    // getter and setter stringType
    public function setStringType($stringType) { $this->stringType = $stringType; }
    public function getStringType() { return $this->stringType; }
}
?>